File: pulsar/java/src/main/java/com/examples/hello/pulsar/consumers/ConsumerController.java
package com.examples.hello.pulsar.consumers;

import com.examples.hello.pulsar.PulsarExecutors;
import com.examples.hello.pulsar.ServiceController;
import io.micronaut.http.annotation.Controller;
import jakarta.inject.Inject;
import jakarta.inject.Named;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.concurrent.ExecutorService;

@Controller(value = "/consumer")
@RequiredArgsConstructor
public class ConsumerController extends ServiceController<ConsumerRunnable> {
  @Inject @Named(PulsarExecutors.CONSUMER) @Getter
  ExecutorService executor;
  @Inject @Getter
  ConsumerRunnable service;
}

File: pulsar/kotlin/src/main/kotlin/com/example/hello/pulsar/consumers/ConsumerController.kt
package com.example.hello.pulsar.consumers

import com.example.hello.pulsar.PulsarExecutors
import com.example.hello.pulsar.ServiceController
import io.micronaut.http.annotation.Controller
import jakarta.inject.Inject
import jakarta.inject.Named
import java.util.concurrent.ExecutorService

@Controller(value = "/consumer")
class ConsumerController(
  @Inject @Named(PulsarExecutors.CONSUMER) val executor: ExecutorService,
  @Inject val service: ConsumerRunnable
) : ServiceController<ConsumerRunnable>(executor, service)
